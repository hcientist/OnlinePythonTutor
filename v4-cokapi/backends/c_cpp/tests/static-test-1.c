void foo(int* a) { // a should point to x
}

int main() {
  static int x[] = {1,2,3,4,5};
  foo(x);
}
