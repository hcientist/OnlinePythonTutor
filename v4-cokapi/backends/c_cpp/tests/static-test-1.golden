{
  "code": "void foo(int* a) { // a should point to x\n}\n\nint main() {\n  static int x[] = {1,2,3,4,5};\n  foo(x);\n}", 
  "trace": [
    {
      "event": "step_line", 
      "func_name": "main", 
      "globals": {}, 
      "heap": {}, 
      "line": 6, 
      "ordered_globals": [], 
      "stack_to_render": [
        {
          "encoded_locals": {
            "x (static 0x601040)": [
              "C_ARRAY", 
              "0x601040", 
              [
                "C_DATA", 
                "0x601040", 
                "int", 
                1
              ], 
              [
                "C_DATA", 
                "0x601044", 
                "int", 
                2
              ], 
              [
                "C_DATA", 
                "0x601048", 
                "int", 
                3
              ], 
              [
                "C_DATA", 
                "0x60104C", 
                "int", 
                4
              ], 
              [
                "C_DATA", 
                "0x601050", 
                "int", 
                5
              ]
            ]
          }, 
          "frame_id": "0xFFF000110", 
          "func_name": "main", 
          "is_highlighted": true, 
          "is_parent": false, 
          "is_zombie": false, 
          "line": 4, 
          "ordered_varnames": [
            "x (static 0x601040)"
          ], 
          "parent_frame_id_list": [], 
          "unique_hash": "main_0xFFF000110"
        }
      ], 
      "stdout": ""
    }, 
    {
      "event": "call", 
      "func_name": "foo", 
      "globals": {}, 
      "heap": {}, 
      "line": 1, 
      "ordered_globals": [], 
      "stack_to_render": [
        {
          "encoded_locals": {
            "x (static 0x601040)": [
              "C_ARRAY", 
              "0x601040", 
              [
                "C_DATA", 
                "0x601040", 
                "int", 
                1
              ], 
              [
                "C_DATA", 
                "0x601044", 
                "int", 
                2
              ], 
              [
                "C_DATA", 
                "0x601048", 
                "int", 
                3
              ], 
              [
                "C_DATA", 
                "0x60104C", 
                "int", 
                4
              ], 
              [
                "C_DATA", 
                "0x601050", 
                "int", 
                5
              ]
            ]
          }, 
          "frame_id": "0xFFF000110", 
          "func_name": "main", 
          "is_highlighted": false, 
          "is_parent": false, 
          "is_zombie": false, 
          "line": 6, 
          "ordered_varnames": [
            "x (static 0x601040)"
          ], 
          "parent_frame_id_list": [], 
          "unique_hash": "main_0xFFF000110"
        }, 
        {
          "encoded_locals": {
            "a": [
              "C_DATA", 
              "0xFFF0000F8", 
              "pointer", 
              "<UNINITIALIZED>"
            ]
          }, 
          "frame_id": "0xFFF000100", 
          "func_name": "foo", 
          "is_highlighted": true, 
          "is_parent": false, 
          "is_zombie": false, 
          "line": 1, 
          "ordered_varnames": [
            "a"
          ], 
          "parent_frame_id_list": [], 
          "unique_hash": "foo_0xFFF000100"
        }
      ], 
      "stdout": ""
    }, 
    {
      "event": "return", 
      "func_name": "foo", 
      "globals": {}, 
      "heap": {}, 
      "line": 2, 
      "ordered_globals": [], 
      "stack_to_render": [
        {
          "encoded_locals": {
            "x (static 0x601040)": [
              "C_ARRAY", 
              "0x601040", 
              [
                "C_DATA", 
                "0x601040", 
                "int", 
                1
              ], 
              [
                "C_DATA", 
                "0x601044", 
                "int", 
                2
              ], 
              [
                "C_DATA", 
                "0x601048", 
                "int", 
                3
              ], 
              [
                "C_DATA", 
                "0x60104C", 
                "int", 
                4
              ], 
              [
                "C_DATA", 
                "0x601050", 
                "int", 
                5
              ]
            ]
          }, 
          "frame_id": "0xFFF000110", 
          "func_name": "main", 
          "is_highlighted": false, 
          "is_parent": false, 
          "is_zombie": false, 
          "line": 6, 
          "ordered_varnames": [
            "x (static 0x601040)"
          ], 
          "parent_frame_id_list": [], 
          "unique_hash": "main_0xFFF000110"
        }, 
        {
          "encoded_locals": {
            "a": [
              "C_DATA", 
              "0xFFF0000F8", 
              "pointer", 
              "0x601040"
            ]
          }, 
          "frame_id": "0xFFF000100", 
          "func_name": "foo", 
          "is_highlighted": true, 
          "is_parent": false, 
          "is_zombie": false, 
          "line": 2, 
          "ordered_varnames": [
            "a"
          ], 
          "parent_frame_id_list": [], 
          "unique_hash": "foo_0xFFF000100"
        }
      ], 
      "stdout": ""
    }, 
    {
      "event": "return", 
      "func_name": "main", 
      "globals": {}, 
      "heap": {}, 
      "line": 7, 
      "ordered_globals": [], 
      "stack_to_render": [
        {
          "encoded_locals": {
            "x (static 0x601040)": [
              "C_ARRAY", 
              "0x601040", 
              [
                "C_DATA", 
                "0x601040", 
                "int", 
                1
              ], 
              [
                "C_DATA", 
                "0x601044", 
                "int", 
                2
              ], 
              [
                "C_DATA", 
                "0x601048", 
                "int", 
                3
              ], 
              [
                "C_DATA", 
                "0x60104C", 
                "int", 
                4
              ], 
              [
                "C_DATA", 
                "0x601050", 
                "int", 
                5
              ]
            ]
          }, 
          "frame_id": "0xFFF000110", 
          "func_name": "main", 
          "is_highlighted": true, 
          "is_parent": false, 
          "is_zombie": false, 
          "line": 7, 
          "ordered_varnames": [
            "x (static 0x601040)"
          ], 
          "parent_frame_id_list": [], 
          "unique_hash": "main_0xFFF000110"
        }
      ], 
      "stdout": ""
    }
  ]
}

